unit_test/fb_serializer_test.cc

```diff
@@ -137,3 +137,35 @@ BOOST_AUTO_TEST_CASE(fb_serializer_generic_event_content_encoding) {
   const auto& batch_metadata = *(event_batch->metadata());
   BOOST_CHECK_EQUAL(batch_metadata.content_encoding()->c_str(), value::CONTENT_ENCODING_DEDUP);
 }
+
+BOOST_AUTO_TEST_CASE(fb_serializer_generic_event_metadata) {
+ data_buffer db;
+ fb_collection_serializer<generic_event> collection_serializer(db, value::CONTENT_ENCODING_DEDUP);
+ const char* event_id("event_id");
+ const timestamp ts;
+ cb_serializer serializer;
+ ranking_response rr(event_id);
+ rr.set_model_id("model_id");
+ rr.push_back(1, 0.2);
+ rr.push_back(0, 0.8);
+
+ auto buffer = serializer.event("my_context", action_flags::DEFERRED, v2::LearningModeType_Apprentice, rr);
+
+ generic_event ge(event_id, ts, v2::PayloadType_CB, std::move(buffer), event_content_type::IDENTITY, "app_id");
+ collection_serializer.add(ge);
+ BOOST_CHECK_EQUAL(reinforcement_learning::error_code::success, collection_serializer.finalize(nullptr));
+
+ flatbuffers::Verifier v(db.body_begin(), db.body_filled_size());
+ const v2::EventBatch *event_batch = v2::GetEventBatch(db.body_begin());
+ BOOST_CHECK(event_batch->Verify(v));
+
+ const auto& events = *(event_batch->events());
+
+ for (size_t i = 0; i < events.size(); ++i) {
+  const auto& serialized_event = *events[(flatbuffers::uoffset_t)i];
+  const auto& payload = serialized_event.payload();
+  const v2::Event *event = flatbuffers::GetRoot<v2::Event>(payload);
+  //const auto& metadata = *(event->meta());
+  //BOOST_CHECK_EQUAL(metadata.app_id()->c_str(), "app_id");
```

Comment:
line 169 in the test throws memory access violation exception. Do you know what I am doing wrong here. It is true of any metadata I am trying access in this test, not just app_id. @olgavrou @ataymano @kumpera

Reply:
Is `meta` valid? Is `app_id` valid?
Try to valid those first for presence before using them.

Reply:
the payload is being accessed and deserialized incorrectly:

```
const auto* serialized_event = events.Get(i);
const v2::Event *event = flatbuffers::GetRoot<v2::Event>(serialized_event->payload()->data());
```
the payload's underlying `data` array needs to be passed into the `GetRoot` deserializer

Reply:
This worked. Thanks a log Olga.